processing/src/main/java/io/druid/segment/data/ColumnarFloats.java

```diff
@@ -33,35 +34,75 @@
 public interface ColumnarFloats extends Closeable
 {
   int size();
+
   float get(int index);
+
   void fill(int index, float[] toFill);
 
   @Override
   void close();
 
-  default ColumnValueSelector<Float> makeColumnValueSelector(ReadableOffset offset)
+  default ColumnValueSelector<Float> makeColumnValueSelector(ReadableOffset offset, ImmutableBitmap nullValueBitmap)
   {
-    class HistoricalFloatColumnSelector implements FloatColumnSelector, HistoricalColumnSelector<Float>
-    {
-      @Override
-      public float getFloat()
+    if (nullValueBitmap.isEmpty()) {
+      class HistoricalFloatColumnSelector implements FloatColumnSelector, HistoricalColumnSelector<Float>
       {
-        return ColumnarFloats.this.get(offset.getOffset());
-      }
+        @Override
+        public float getFloat()
+        {
+          return ColumnarFloats.this.get(offset.getOffset());
+        }
 
-      @Override
-      public double getDouble(int offset)
-      {
-        return ColumnarFloats.this.get(offset);
-      }
+        @Override
```

Comment:
not sure how this helps, but still rearranged.

Reply:
When "specialization" classes (Long/Float/Double) are identical except the "Long/Float/Double" word, it's easier to review changes, to apply changes by analogy, and it doesn't raise any questions. When they are different, it's always a question if they are different because there is some specific logic of a specific type (sometimes there is), or for no actual reason.